### ISSUE TITLE
No doc for localize

### ISSUE THREAD

**Issue by PeterWone** [microsoft/vscode-extension-samples#481]
There's a sample but it's worthless if you can't read Japanese because who in their right mind would set the display language to a language they can't read while trying to figure out a library?

I'd be able to use it if the sample language were French but that still wouldn't be any use to people who can't read French.

FFS write one wiki page explaining what's where and the relationship between the symbols and where they are. 

I tried looking at the source code for the localize function but somebody wrote super clever code that dynamically overloads methods making it very hard to follow. Do NOT refer me to Stack Overflow, I asked about this two years ago and have not received so much as a comment.

**Comment by dbaeumer** [comment]
@PeterWone thanks for detailed suggestions.

**Comment by PeterWone** [comment]
Dirk, sorry about the tone. I was getting pretty frustrated. I never did figure out how webpack is interfering. Eventually I wrote my own code to pull values out of the generated files.

```ts
import { env } from 'vscode';
const locKey = require("../out/extension.nls.metadata.json").keys;
const locVal: any = {
  "de": require("../out/extension.nls.de.json"),
  "en": require("../out/extension.nls.en.json"),
  "fr": require("../out/extension.nls.fr.json")
};

export function localise(key: string): string {
  const i: number = locKey.indexOf(key);
  let translations = locVal[env.language] || locVal["en"];
  return i == -1 ? key : translations[i];
}
```

**Comment by PeterWone** [comment]
Re the sample: maintaining 200 language versions is untenable, and English is the lingua franca for development, so probably the simplest solution is to use as the target language a language that supports the Roman alphabet, and then use mock French "translated" strings that are the English strings with (FR) prepended so you can see that translation has occurred. Once that's in place people who care enough can PR versions for their own language and character set.
